packages/dependency-extraction-webpack-plugin/package.json

```diff
@@ -19,7 +19,8 @@
 		"url": "https://github.com/WordPress/gutenberg/issues"
 	},
 	"files": [
-		"index.js"
+		"index.js",
+		"util.js"
```

Comment:
I believe we could `*.js` to include all JS files in the root rather than listing each one.

https://docs.npmjs.com/files/package.json#files

Reply:
Whatever the preferences are for the project :-) I guess for this package with relatively low maintenance it shouldn‘t matter too much.

Reply:
> I believe we could `*.js` to include all JS files in the root rather than listing each one.

I'm not sure about this. Part of the intention is to omit development-only files, like test files. Using `*.js` would probably include those.

At a certain point, it begs the question whether we should be using these whitelists at all. They can definitely help in some cases where local developer files can be accidentally included in published packages (sometimes with catastrophic consequences, in the case of secrets files).

Reply:
At this point, I think the proposed changes should suffice. We can revisit it at the project level if it continues to be a problem.

Reply:
> I'm not sure about this. Part of the intention is to omit development-only files, like test files. Using *.js would probably include those.

We can actually verify this by entering the package directory and running `npm publish --dry-run`. The current version or `*.js` should work.


As is: 
```
npm notice 📦  @wordpress/dependency-extraction-webpack-plugin@1.0.0
npm notice === Tarball Contents ===
npm notice 833B  package.json
npm notice 115B  CHANGELOG.md
npm notice 4.0kB index.js
npm notice 6.8kB README.md
npm notice 2.0kB util.js
npm notice === Tarball Details ===
npm notice name:          @wordpress/dependency-extraction-webpack-plugin
npm notice version:       1.0.0
npm notice package size:  4.7 kB
npm notice unpacked size: 13.8 kB
npm notice shasum:        684f1054a2f3193e2202e042e7029b1205852271
npm notice integrity:     sha512-lQX+jxfimL90s[...]rdPP8LtptGyNQ==
npm notice total files:   5
npm notice
```

with `*.js`

```
npm notice
npm notice 📦  @wordpress/dependency-extraction-webpack-plugin@1.0.0
npm notice === Tarball Contents ===
npm notice 816B  package.json
npm notice 115B  CHANGELOG.md
npm notice 4.0kB index.js
npm notice 6.8kB README.md
npm notice 2.0kB util.js
npm notice === Tarball Details ===
npm notice name:          @wordpress/dependency-extraction-webpack-plugin
npm notice version:       1.0.0
npm notice package size:  4.7 kB
npm notice unpacked size: 13.8 kB
npm notice shasum:        d3f10e0035e2fe0d8b09072289dd63ba4cf299ee
npm notice integrity:     sha512-rFGZzbiLhQmUY[...]LJlXdp6Yx9W7A==
npm notice total files:   5
npm notice
```

---

> At a certain point, it begs the question whether we should be using these whitelists at all.

It was very unfortunate to publish a broken version of this package. The appeal of the whitelist is that this package is very small. Without the whitelist it's still very small, but includes a lot of irrelevant files.

```
npm notice
npm notice 📦  @wordpress/dependency-extraction-webpack-plugin@1.0.0
npm notice === Tarball Contents ===
npm notice 791B  package.json
npm notice 115B  CHANGELOG.md
npm notice 4.0kB index.js
npm notice 6.8kB README.md
npm notice 2.0kB util.js
npm notice 4.6kB test/__snapshots__/build.js.snap
npm notice 2.2kB test/build.js
npm notice 93B   test/fixtures/dynamic-import/index.js
npm notice 93B   test/fixtures/dynamic-import/util.js
npm notice 143B  test/fixtures/dynamic-import/webpack.config.js
npm notice 168B  test/fixtures/function-output-filename/index.js
npm notice 246B  test/fixtures/function-output-filename/webpack.config.js
npm notice 83B   test/fixtures/no-default/index.js
npm notice 167B  test/fixtures/no-default/webpack.config.js
npm notice 24B   test/fixtures/no-deps/index.js
npm notice 143B  test/fixtures/no-deps/webpack.config.js
npm notice 340B  test/fixtures/overrides/index.js
npm notice 487B  test/fixtures/overrides/webpack.config.js
npm notice 340B  test/fixtures/with-externs/index.js
npm notice 245B  test/fixtures/with-externs/webpack.config.js
npm notice 182B  test/fixtures/wordpress-require/index.js
npm notice 143B  test/fixtures/wordpress-require/webpack.config.js
npm notice 168B  test/fixtures/wordpress/index.js
npm notice 143B  test/fixtures/wordpress/webpack.config.js
npm notice 1.2kB test/util.js
npm notice === Tarball Details ===
npm notice name:          @wordpress/dependency-extraction-webpack-plugin
npm notice version:       1.0.0
npm notice package size:  7.0 kB
npm notice unpacked size: 25.0 kB
npm notice shasum:        5ab304a0954aada1ead523c5caa4b9c2e6a89ba8
npm notice integrity:     sha512-C/ikDJ7sETOLi[...]n7i3s64EyxMQA==
npm notice total files:   25
npm notice
```

Reply:
> We can actually verify this by entering the package directory and running `npm publish --dry-run`. The current version or `*.js` should work.

I guess I expected `*.js` would match deeply in folders. From your results, it seems to only look in the (package's) top-level directory?

Reply:
The [documentation on `files`](https://docs.npmjs.com/files/package.json#files) says it uses a gitignore-like syntax with globs.

`*` matches anything except `/`, so `*.js` will only match top level files with the `.js` extension.

Reply:
We probably should tackle `files` in all `package.json` in one go to bring some consistency. Let's merge this PR  as is to unblock those using `@wordpress/scripts` and figure out the best path forward separately.